### PR TITLE
Update webpack

### PR DESCRIPTION
**Pull request by MGibson1**
## Type of change

- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective
Update Webpack

## Code changes
* **package-lock**: generated from changes to `package.json`
* **package.json**: update webpack and plugins to latest, match engine to other projects
* **webpack.config.js**: comply to new webpack api

## Testing requirements
No changes expected, general regression testing should suffice.

## Before you submit

- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -35,7 +35,10 @@ const plugins = [
     banner: "#!/usr/bin/env node",
     raw: true,
   }),
-  new webpack.IgnorePlugin(/^encoding$/, /node-fetch/),
+  new webpack.IgnorePlugin({
+    resourceRegExp: /^encoding$ /,
```
Comment (djsmith85) [inline webpack.config.js:39]: Still a space at the end of the regex. Not sure if it has an impact, as the build seems to work fine.

File: package.json [hunk 2]
```diff
@@ -97,7 +98,8 @@
     "zxcvbn": "^4.4.2"
   },
   "engines": {
-    "node": ">=16"
+    "node": "~16",
+    "npm": "~8"
```
Comment (djsmith85) [inline package.json:102]: The change to node >=16 was intentional as you can see here: https://github.com/bitwarden/cli/pull/422#discussion_r767603984

**Review by djsmith85 — CHANGES_REQUESTED** [review]
Just some minor things. I have no idea why `rimraf` needed to be added as we don't have it on any other repos as a dev-dependency.

Rest looks good.

File: webpack.config.js (hunk 1 above)
Comment (MGibson1) [reply webpack.config.js:39]: lol embarrassing

File: package.json (hunk 2 above)
Comment (MGibson1) [reply package.json:102]: we did this specifically to cause a warning. The intention is to indicate that you are building in a system we have not tested functionality on.

**Review by djsmith85 — APPROVED** [review]

**Pull request merged**